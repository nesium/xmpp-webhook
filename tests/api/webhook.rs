use anyhow::Result;
use insta::assert_snapshot;

use crate::helpers::spawn_app;

#[tokio::test]
async fn test_push() -> Result<()> {
    let app = spawn_app().await;
    let client = reqwest::Client::new();

    let response = client
        .post(&format!("{}/webhook", &app.address))
        .header("X-GitHub-Event", "push")
        .header("Content-Type", "application/json")
        .body(include_str!("fixtures/push.json"))
        .send()
        .await?;

    assert!(response.status().is_success());
    assert_snapshot!(app.xmpp.sent_messages()[0].message);

    Ok(())
}

#[tokio::test]
async fn test_issue_opened() -> Result<()> {
    let app = spawn_app().await;
    let client = reqwest::Client::new();

    let response = client
        .post(&format!("{}/webhook", &app.address))
        .header("X-GitHub-Event", "issues")
        .header("Content-Type", "application/json")
        .body(include_str!("fixtures/issue_opened.json"))
        .send()
        .await?;

    assert!(response.status().is_success());
    assert_snapshot!(app.xmpp.sent_messages()[0].message);

    Ok(())
}

#[tokio::test]
async fn test_issue_comment_created() -> Result<()> {
    let app = spawn_app().await;
    let client = reqwest::Client::new();

    let response = client
        .post(&format!("{}/webhook", &app.address))
        .header("X-GitHub-Event", "issue_comment")
        .header("Content-Type", "application/json")
        .body(include_str!("fixtures/issue_comment_created.json"))
        .send()
        .await?;

    assert!(response.status().is_success());
    assert_snapshot!(app.xmpp.sent_messages()[0].message);

    Ok(())
}
