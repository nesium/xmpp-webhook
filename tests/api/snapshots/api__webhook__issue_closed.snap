---
source: tests/api/webhook.rs
expression: "sent_messages[0].message"
---
[nesium](https://github.com/nesium) has closed [issue #2](https://github.com/nesium/test_repo/issues/2) in [test_repo](https://github.com/nesium/test_repo)

**Title**: Item 1
